src/features/smartSearch/components/filters/EmailHistory/index.tsx

```diff
@@ -0,0 +1,208 @@
+import { MenuItem } from '@mui/material';
+import { useState } from 'react';
+
+import convertToMessageKey from './convertToMessageKey';
+import { EMAIL_SELECT_SCOPE } from '../EmailClick';
+import FilterForm from '../../FilterForm';
+import messageIds from 'features/smartSearch/l10n/messageIds';
+import { Msg } from 'core/i18n';
+import StyledSelect from '../../inputs/StyledSelect';
+import TimeFrame from '../TimeFrame';
+import useCampaigns from 'features/campaigns/hooks/useCampaigns';
+import useEmails from 'features/emails/hooks/useEmails';
+import { useNumericRouteParams } from 'core/hooks';
+import useSmartSearchFilter from 'features/smartSearch/hooks/useSmartSearchFilter';
+import {
+  EmailHistoryFilterConfig,
+  NewSmartSearchFilter,
+  OPERATION,
+  SmartSearchFilterWithId,
+  ZetkinSmartSearchFilter,
+} from 'features/smartSearch/components/types';
+
+const localMessageIds = messageIds.filters.emailHistory;
+
+enum EMAIL_HISTORY_OP {
+  OPENED = 'opened',
+  NOT_OPENED = 'not_opened',
+  NOT_SENT = 'not_sent',
+  SENT = 'sent',
+}
+
+interface EmailHistoryProps {
+  filter:
+    | SmartSearchFilterWithId<EmailHistoryFilterConfig>
+    | NewSmartSearchFilter;
+  onSubmit: (
+    filter:
+      | SmartSearchFilterWithId<EmailHistoryFilterConfig>
+      | ZetkinSmartSearchFilter<EmailHistoryFilterConfig>
+  ) => void;
+  onCancel: () => void;
+}
+
+const EmailHistory = ({
+  filter: initialFilter,
+  onCancel,
+  onSubmit,
+}: EmailHistoryProps): JSX.Element => {
+  const { orgId } = useNumericRouteParams();
+  const emailsFuture = useEmails(orgId).data || [];
+  const projectsFuture = useCampaigns(orgId).data || [];
+
+  const { filter, setConfig, setOp } =
+    useSmartSearchFilter<EmailHistoryFilterConfig>(initialFilter, {
+      operator: 'sent',
+    });
+  const [emailSelectScope, setEmailSelectScope] = useState<EMAIL_SELECT_SCOPE>(
+    filter.config.campaign
+      ? EMAIL_SELECT_SCOPE.FROM_PROJECT
+      : filter.config.email
+      ? EMAIL_SELECT_SCOPE.SPECIFIC_EMAIL
+      : EMAIL_SELECT_SCOPE.ANY
+  );
+
+  const setValueToKey = (
+    key: keyof EmailHistoryFilterConfig,
+    value: string | number
+  ) => {
+    setConfig({
+      ...filter.config,
+      [key]: value,
+    });
+  };
+  const removeKey = (deleteKeys: (keyof EmailHistoryFilterConfig)[]) => {
+    const copied = { ...filter.config };
+    deleteKeys.forEach((key) => delete copied[key]);
+    setConfig({
+      ...copied,
+    });
+  };
+
+  const handleTimeFrameChange = (range: {
+    after?: string;
+    before?: string;
+  }) => {
+    /* eslint-disable-next-line */
```

Comment:
I don't like when linting rules are disabled, and in this case I didn't understand the reason, so I tried removing this comment. It took me a while to understand what was going on here. Is this something you've come up with or did you copy it from somewhere else? I think it would be good to explain it in a comment, or perhaps creating a utility function in order to name the operation that is taking place here.